source/jormungandr/tests/kirin_realtime_tests.py

```diff
@@ -2196,6 +2196,370 @@ def test_add_new_trip_present_in_navitia_with_a_shift(self):
+        #     the day before,
+        #     the very day (20120616)
+        #     and the day after the disruption
+        # > vjA circulates everyday with departure at 08:01:00 and arrival at 08:01:02
+        ba_15T08_journey_query = empty_query.format(
+            f='stop_point:stopB', to='stop_point:stopA', dt='20120615T080000'
+        )
+        response = self.query_region(ba_15T08_journey_query)
+        assert len(response['journeys']) == 1
+        assert response['journeys'][0]['departure_date_time'] == '20120615T080100'
+        assert response['journeys'][0]['arrival_date_time'] == '20120615T080102'
+        assert response['journeys'][0]['sections'][0]['display_informations']['name'] == 'A'
+
+        ba_16T08_journey_query = empty_query.format(
+            f='stop_point:stopB', to='stop_point:stopA', dt='20120616T080000'
+        )
+        response = self.query_region(ba_16T08_journey_query)
+        assert len(response['journeys']) == 1
+        assert response['journeys'][0]['departure_date_time'] == '20120616T080100'
+        assert response['journeys'][0]['arrival_date_time'] == '20120616T080102'
+        assert response['journeys'][0]['sections'][0]['display_informations']['name'] == 'A'
+
+        ba_17T08_journey_query = empty_query.format(
+            f='stop_point:stopB', to='stop_point:stopA', dt='20120617T080000'
+        )
+        response = self.query_region(ba_17T08_journey_query)
+        assert len(response['journeys']) == 1
+        assert response['journeys'][0]['departure_date_time'] == '20120617T080100'
+        assert response['journeys'][0]['arrival_date_time'] == '20120617T080102'
+        assert response['journeys'][0]['sections'][0]['display_informations']['name'] == 'A'
+
+        # Also check journeys in realtime from C to A at 22:00
+        #     the day before,
+        #     the very day (20120615 as it's a "anti" pass-midnight)
+        #     and the day after the disruption
+        # > No result before the disruption
+        ba_14T22_journey_query = empty_query.format(
+            f='stop_point:stopC', to='stop_point:stopA', dt='20120614T220000'
+        )
+        response, status = self.query_region(ba_14T22_journey_query, check=False)
+        assert 'journeys' not in response
+
+        ba_15T22_journey_query = empty_query.format(
+            f='stop_point:stopC', to='stop_point:stopA', dt='20120615T220000'
+        )
+        response, status = self.query_region(ba_15T22_journey_query, check=False)
+        assert 'journeys' not in response
+
+        ba_16T22_journey_query = empty_query.format(
+            f='stop_point:stopC', to='stop_point:stopA', dt='20120616T220000'
+        )
+        response, status = self.query_region(ba_16T22_journey_query, check=False)
+        assert 'journeys' not in response
+
+        # A new disruption with a detour on departure station to have a pass midnight in advance
+        self.send_mock(
+            "vjA",
+            "20120615",
```

Comment:
This should probably be 20120616 (as it's supposed to be GTFS base-VJ's date in GTFS-RT's doc), but then the disruption seems to be rejected by Kraken.